persistence-test/src/main/java/io/stargate/it/storage/ExternalStorage.java

```diff
@@ -497,4 +499,141 @@ public void start() {
+    private static final String DSE_VERSION = System.getProperty("dse.version");
+    private static final String DSE_REPOSITORY =
+        System.getProperty("dse.repository", "datastax/dse-server");
+
+    private static final String CASSANDRA_VERSION = System.getProperty("cassandra.version");
+    private static final String CASSANDRA_REPOSITORY =
+        System.getProperty("cassandra.repository", "cassandra");
+
+    private final List<GenericContainer> nodes;
+    private final String initSite;
+    private final AtomicBoolean removed = new AtomicBoolean();
+
+    private final String broadcastAddress = "host.testcontainers.internal";
+
+    public DockerCluster(ClusterSpec spec, ExtensionContext context) {
+      super(spec);
+      this.initSite = context.getUniqueId();
+
+      ImmutableList.Builder<GenericContainer> nodes = ImmutableList.builder();
+      for (int i = 0; i < spec.nodes(); i++) {
+        nodes.add(createCassandraContainer(i));
+      }
+      this.nodes = nodes.build();
+
+      Testcontainers.exposeHostPorts(7000);
+    }
+
+    @Override
+    public String seedAddress() {
+      // return "127.0.0.1";
+      return nodes.get(0).getHost();
+    }
+
+    @Override
+    public int storagePort() {
+      return nodes.get(0).getMappedPort(7000);
+    }
+
+    @Override
+    public int cqlPort() {
+      return nodes.get(0).getMappedPort(9042);
+    }
+
+    @Override
+    public String clusterVersion() {
+      return isDse() ? DSE_VERSION : CASSANDRA_VERSION;
+    }
+
+    @Override
+    public boolean isDse() {
+      return DSE_VERSION != null;
+    }
+
+    @Override
+    public void start() {
+      if (!EXTERNAL_BACKEND) {
+        ShutdownHook.add(this);
+
+        for (GenericContainer node : nodes) {
+          node.start();
+        }
+
+        LOG.info(
+            "Storage cluster requested by {} has been started with version {}",
+            initSite,
+            clusterVersion());
+      }
+    }
+
+    public void close() {
+      super.close();
+      stop();
+    }
+
+    public void stop() {
+      if (!EXTERNAL_BACKEND) {
+        ShutdownHook.remove(this);
+
+        try {
+          if (removed.compareAndSet(false, true)) {
+
+            for (GenericContainer node : nodes) {
+              node.stop();
+            }
+
+            LOG.info(
+                "Storage cluster (version {}) that was requested by {} has been removed.",
+                clusterVersion(),
+                initSite);
+          }
+        } catch (Exception e) {
+          // This should not affect test result validity, hence logging as WARN
+          LOG.warn("Exception during Docker cluster shutdown: " + e.getMessage(), e);
+        }
+      }
+    }
+
+    private GenericContainer<?> createCassandraContainer(int num) {
+      String image =
+          isDse()
+              ? DSE_REPOSITORY + ":" + DSE_VERSION
+              : CASSANDRA_REPOSITORY + ":" + CASSANDRA_VERSION;
+
+      GenericContainer<?> container =
```

Comment:
* what about the seeds? like set seed address for the ones that have the `num > 0`..
* I think you should also define the alias for each node, like `cassandra-0`, `cassandra-1`, etc